packages/test/package.json

```
@@ -3,24 +3,40 @@
   "version": "3.0.1",
   "description": "Testing utilities for Rest Hooks",
   "sideEffects": false,
-  "main": "dist/index.cjs.js",
-  "module": "lib/index.js",
+  "main": "dist/index.cjs",
+  "module": "legacy/index.js",
   "types": "lib/index.d.ts",
+  "type": "module",
+  "exports": {
+    "node": {
+      "module": "./dist/index.js",
+      "require": "./dist/index.cjs",
+      "types": "./lib/index.d.ts"
+    },
+    "types": "./lib/browser.d.ts",
+    "default": "./lib/browser.js"
+  },
```

```suggestion
  "exports": {
    ".": [
      {
        "node": {
          "module": "./dist/index.js",
          "require": "./dist/index.cjs",
          "types": "./lib/index.d.ts"
        },
        "types": "./lib/browser.d.ts",
        "default": "./lib/browser.js"
      },
      "./lib/browser.js",
    ]
  }
```
the "default" must be CJS; you need the array-to-string fallback for node v13.0-v13.6.

i have no idea what your intention is here tho - are the entry points "node" and "types" and the main? i'm not familiar with putting a "types" field anywhere inside "exports".

---

These are conditions - using https://webpack.js.org/guides/package-exports/#reference-syntax as my guide. Is there a node reference for 'exports'? It's hard to find anything on google since exports is also the name of normal es6 module syntax.

According to this: https://webpack.js.org/guides/package-exports/#abbreviation the ".": part isn't necessary

"the "default" must be CJS; you need the array-to-string fallback for node v13.0-v13.6." - i'm really confused by this sentence. is there a doc that explains this fallback? this seems like nonsequitor. also, why would default need to be CJS? do older versions of node secretly support 'exports'?

---

the `.` part is necessary if you want to add additional entry points, and you need to add `package.json` in here explicitly or some tooling will break.

`default` should match `main`, so that `require()`  of your package in node v12.16 is the same thing as in v12.17.

---

ahh, good point about keeping package.json.
does node 12.17 support 'exports' field? I thought this was relatively new so it required 14 or something?

I guess an open question from me - is there a version of node that supports exports, but not conditionals inside? Do you know where docs are for how node processes exports and what conditionals it might look at?

---

v12.17+, v13.7+, and v14+ support everything.

node v13.0 and v13.1 support "exports" but only the string form; v13.2-v13.6 supports the object form but only the "default" condition.

There are no docs I'm aware of that describe multiple node versions.